vmdb/app/controllers/mixins/vm_show_mixin.rb

```diff
@@ -26,7 +26,11 @@ def explorer
     end
 
     # Build the Explorer screen from scratch
-    build_trees_and_accordions
+    allowed_features = features.select { |f| role_allows(:feature => f.role) }
+    allowed_features.each { |feature| build_vm_tree(feature.name, feature.tree_name) }
```

Comment:
I *really* don't like this line because it's essentially just for side effects, but that's what the code did before.